service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/devicetwin/QueryCollection.java

```diff
@@ -116,14 +127,86 @@ protected QueryCollection(int pageSize, QueryType requestQueryType, IotHubConnec
         this.isInitialQuery = true;
     }
 
+    /**
+     * Constructor for sql based queries
+     *
+     * @param query the sql query to use.
+     * @param pageSize the size of the page to return per query collection response.
+     * @param requestQueryType the type of query this is.
+     * @param iotHubConnectionString the connection string to connect with to query against.
+     * @param url the url to query against.
+     * @param httpMethod the http method to call with the query.
+     * @param httpConnectTimeout the http connect timeout to use.
+     * @param httpReadTimeout the http read timeout to use.
+     * @throws IllegalArgumentException if page size is 0 or negative, or if the query type is null or unknown, of if the query string is null or empty,
+     *  or if the provided connection string is null, or if the provided url is null, or if the provided http method is null.
+     */
+    protected QueryCollection(String query, int pageSize, QueryType requestQueryType, IotHubConnectionString iotHubConnectionString, URL url, HttpMethod httpMethod, int httpConnectTimeout, int httpReadTimeout, Proxy proxy)
+    {
+        this.validateQueryRequestArguments(iotHubConnectionString, url, httpMethod, pageSize, requestQueryType);
+
+        ParserUtility.validateQuery(query);
+
+        this.pageSize = pageSize;
+        this.query = query;
+        this.requestQueryType = requestQueryType;
+        this.iotHubConnectionString = iotHubConnectionString;
+        this.responseContinuationToken = null;
+        this.httpMethod = httpMethod;
+        this.httpConnectTimeout = httpConnectTimeout;
+        this.httpReadTimeout = httpReadTimeout;
+        this.proxy = proxy;
+        this.url = url;
+        this.responseQueryType = QueryType.UNKNOWN;
+
+        this.isSqlQuery = true;
+
+        this.isInitialQuery = true;
+    }
+
+    /**
+     * Constructor for non-sql based queries.
```

Comment:
Q - haven't seen this before! Do we have a sample/ test on how these non-sql queries are used?

Reply:
Honestly, I'm not sure how this API gets used. "Non-sql query" seems contradictory to me